standard/PortalStatistics.lua

```diff
@@ -0,0 +1,1138 @@
+				return Logic.readBool(isEmptyCategory[catIndex]) end)
+
+		seriesData = Array.map(seriesData, function(_, index)
+				return Array.filter(seriesData[index], function(_, catIndex)
+						return Logic.readBool(isEmptyCategory[catIndex]) end)
+			end)
+	end
+
+	for seriesIndex, series in pairs(seriesNames) do
+		if config.removeEmptySeries == true and Array.all(seriesData[seriesIndex], function(value)
+			return value == 0
+		end) then
+			mw.logObject(series .. ' is empty')
+		else
+			table.insert(chartData, {
+					['name'] = series,
+					['type'] = 'bar',
+					['stack'] = 'total',
+					['data'] = seriesData[seriesIndex],
+					['emphasis'] = {['focus'] = 'series'},
+				}
+			)
+		end
+	end
+
+	config['yAxis'] = {type = 'value', name = 'Earnings ($USD)'}
+	config['xAxis'] = {type = 'category', name = config.catLabel, data = categoryNames}
+	config['customLegend'] = config['customLegend'] or seriesNames
+
+	return StatisticsPortal._drawChart(config, chartData)
+end
+
+
+function StatisticsPortal._chartConfig(args, params)
+
+	local isForTeam = String.isNotEmpty(args.team) or Logic.readBool(args.isForTeam)
+
+	return {
+		processFunction = params.processFunction or nil,
+		variable = params.variable or nil,
+		catLabel = params.catLabel,
+		flipAxes = params.flipAxes or false,
+
+		customLegend = String.isNotEmpty(args.customLegend) and
+			Array.map(mw.text.split(args.customLegend, ','), String.trim) or args.customInputs,
+		customInputs = args.customInputs,
+		customYears = args.customYears,
+		startYear = args.startYear or Info.startYear,
+		yearBreakdown = Logic.readBoolOrNil(args.yearBreakdown),
+		removeEmptyCategories = Logic.readBool(args.removeEmptyCategories) or false,
+		removeEmptySeries = Logic.readBool(args.removeEmptySeries) or false,
+		isForTeam = isForTeam,
+		opponentName = isForTeam and args.team or args.player,
+		opponentType = isForTeam and Opponent.team or Opponent.solo,
+		height = args.height or 400,
+		width = args.width or (200 + 65 * (CURRENT_YEAR - (args.startYear or Info.startYear))),
+	}
+end
+
+
+function StatisticsPortal._toOpponent(player)
+	return {
+		type = Opponent.solo,
+		players = {{
+			pageName = player.pagename,
+			displayName = player.id,
+			flag = player.nationality,
+			team = String.isNotEmpty(player.team) and player.team or nil,
+		}},
+	}
+end
+
+
+function StatisticsPortal._returnCustomYears(args)
+
+	args.startYear = tonumber(args.startYear) or Info.startYear
+	local yearTable
+	local defaultYearTable = Array.range(args.startYear, CURRENT_YEAR)
+
+	if String.isNotEmpty(args.customYears) then
+		yearTable = Array.map(mw.text.split(args.customYears .. ',' .. CURRENT_YEAR, ','), function(item)
+			return tonumber(String.trim(item))
+		end)
+		return yearTable, defaultYearTable
+	else
+		return defaultYearTable, defaultYearTable
+	end
+end
+
+
+function StatisticsPortal._returnCustomYearText(prevYear, yearValue)
+	return (prevYear == yearValue) and tostring(yearValue) or
```

Comment:
is the tostring needed?

Reply:
The values are numbers passed to the function are numbers

Reply:
you can concat numbers and strings without having to tostring the numbers